.github/workflows/scripts/detect_ts_ignore.py

```
@@ -0,0 +1,37 @@
+    ts_ignore_found = False
+
+    for file in files:
+        try:
+            with open(file, encoding="utf-8") as f:
+                for line_num, line in enumerate(f, start=1):
+                    if "@ts-ignore" in line:
+                        print(f"❌ Error: '@ts-ignore' found in {file} at line {line_num}")
+                        ts_ignore_found = True
+        except FileNotFoundError:
+            print(f"⚠️ Warning: File not found: {file}")
+        except (IOError, OSError) as e:
```

Ruff (0.8.2)

Replace aliased errors with `OSError`

Replace with builtin `OSError`

(UP024)